### PR TITLE
Categorize entries in person-person relations select (#1390)

### PR DESCRIPTION
**Pull request by blms**
## In this PR

Per #1390:
- Use `select` and `optgroup` to visually categorize (and alphabetize) person-person entity relations in Person form
- Per PGP-Performant meetings, these options are meant to be more or less set in stone, unless there is a significant need for a new/edited option, so the add/remove/change buttons are hidden from the inline

### CODE REVIEW

File: geniza/entities/admin.py [hunk 1]
```diff
@@ -132,7 +172,13 @@ def get_form(self, request, obj=None, **kwargs):
...
         if obj:
             self.own_pk = obj.pk
-        return super().get_form(request, obj, **kwargs)
+
+        # remove add/change/delete buttons from "role" widget
+        form = super().get_form(request, obj, **kwargs)
+        form.base_fields["role"].widget.can_add_related = False
+        form.base_fields["role"].widget.can_change_related = False
+        form.base_fields["role"].widget.can_delete_related = False
+        return form
```
Comment (rlskoeser) [inline geniza/entities/admin.py:181]: would it be better to handle by user permissions, limiting which user groups have access to add/change/delete these?

**Review by rlskoeser — APPROVED** [review]
Seems reasonable, just wonder if it would be better to handle permissions at user/group level rather than overriding here.

File: geniza/entities/admin.py (hunk 1 above)
Comment (blms) [reply geniza/entities/admin.py:181]: Good question, asked the team! I know originally their decision was to make this editable only from the admin list view, and not from the inlines, but I've asked if they'd prefer group-based permissions.

Comment (blms) [reply geniza/entities/admin.py]: Thanks for raising this, the team agrees group-based permissions are preferable!

**Comment by blms** [comment]
@rlskoeser Going to go ahead and merge since all has been reviewed except the migration, and it's the same standard group permissions migration (including the `create_permissions` call!) that we've used in the past.

**Pull request merged**